relays/ethereum/src/message_race_delivery.rs

```diff
@@ -0,0 +1,385 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+//! Message delivery race delivers proof-of-messages from lane.source to lane.target.
+
+use crate::message_lane::{MessageLane, SourceHeaderIdOf, TargetHeaderIdOf};
+use crate::message_lane_loop::{
+	SourceClient as MessageLaneSourceClient, SourceClientState, TargetClient as MessageLaneTargetClient,
+	TargetClientState,
+};
+use crate::message_race_loop::{MessageRace, RaceState, RaceStrategy, SourceClient, TargetClient};
+use crate::utils::FailedClient;
+
+use async_trait::async_trait;
+use futures::stream::FusedStream;
+use std::{collections::VecDeque, marker::PhantomData, ops::RangeInclusive, time::Duration};
+
+/// Maximal number of messages to relay in single transaction.
+const MAX_MESSAGES_TO_RELAY_IN_SINGLE_TX: u32 = 4;
+
+/// Run message delivery race.
+pub async fn run<P: MessageLane>(
```

Comment:
This is basically just a pass-through to `message_race_loop`, so why not call it from `message_lane_loop` directly?

Reply:
It is exposing 4 private types vs exposing 1 function. I'm for the latter